contracts/serverless-v2/GatewayJobs.sol

```diff
@@ -242,20 +243,26 @@ contract GatewayJobs is
         uint256 _usdcDeposit,
         uint8 _sequenceId
     ) internal {
-        (uint256 execJobId, uint8 errorCode) = jobMgr.createJob(_codehash, _codeInputs, _deadline);
-        if (errorCode == 1) {
-            // Resource unavailable
-            relayJobs[_jobId].isResourceUnavailable = true;
-            return;
-        }
 
-        relayJobs[_jobId].execStartTime = block.timestamp;
-        relayJobs[_jobId].jobOwner = _jobOwner;
-        relayJobs[_jobId].usdcDeposit = _usdcDeposit;
-        relayJobs[_jobId].sequenceId = _sequenceId;
-        relayJobs[_jobId].gateway = _gateway;
-        execJobs[execJobId] = _jobId;
-        emit JobCreated(_jobId, execJobId, _jobOwner, _gateway);
+        try jobMgr.createJob(_codehash, _codeInputs, _deadline) returns (uint256 execJobId) {
+            relayJobs[_jobId].execStartTime = block.timestamp;
+            relayJobs[_jobId].jobOwner = _jobOwner;
+            relayJobs[_jobId].usdcDeposit = _usdcDeposit;
+            relayJobs[_jobId].sequenceId = _sequenceId;
+            relayJobs[_jobId].gateway = _gateway;
+            execJobs[execJobId] = _jobId;
+            emit JobCreated(_jobId, execJobId, _jobOwner, _gateway);
+        } catch (bytes memory reason) {
+            if (keccak256(reason) == keccak256(abi.encodePacked(Jobs.JobsUnavailableResources.selector))) {
```

Comment:
maybe we can compute the RHS offchain and hardcode here

Reply:
declared the value as constant, to be computed at compile time only.

Reply:
any reason to hash it on both sides?

Reply:
the way they do it here seems more in line with what I would expect - https://dev.to/george_k/embracing-custom-errors-in-solidity-55p8

Reply:
Above article is suggesting to use following:

`bytes4 errorSelector = abi.decode(reason, (bytes4));`

But it has one downside, this will panic if reason is not of bytes4. And the `reason` will be lost.

Reason for using hash on both side:
reason is of type bytes, can't be sliced to bytes4. And to check  equality of two bytes variable, == doesn't work. So hashing both side seems one option.

Reply:
Updated: using cast to bytes4  on reason to compare with error selector.